<div style="border:solid Chocolate 2px; padding: 40px">
 
<b> Глеб, привет!👋</b> 

Меня зовут Кирилл Васильев, я буду ревьюером твоего проекта. Я предлагаю общаться на «ты», но если привычнее на «вы», просто скажи об этом! 

Я буду оставлять комментарии в твоем проекте. Пожалуйста, не удаляй их, они могут быть полезны в случае повторной проверки.

Свои комментарии я буду обозначать <font color='green'>зеленым</font>, <font color='gold'>желтым</font> и <font color='red'>красным</font> цветами, например:
  
<br/> 

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все отлично!👍:</b> В случае, если решение на отдельном шаге является полностью правильным. Здесь же я могу давать советы и предложения.
</div>

<br/>

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> В случае, когда решение на отдельном шаге станет еще лучше, если внести небольшие коррективы.
</div> 
  

<br/>
<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера <a class="tocSkip"></h2>
 
   
<b>На доработку🤔:</b>
 В случае, когда решение на отдельном шаге требует существенной переработки и внесения правок. Напоминаю, что проект не может быть принят с первого раза, если ревью содержит комментарии, рекомендующие доработать шаги.
</div>  

<br/>
<div class="alerфt alert-info">
Рекомендую для твоих комментариев использовать синий или любой другой цветной фон. Я постараюсь ничего не пропустить, но если комментарии будут выделены, то это мне очень поможет!
</div>
    
Увидев у тебя неточность, в первый раз я лишь укажу на ее наличие и дам тебе возможность самостоятельно найти и исправить ее. На реальной работе твой руководитель будет поступать также. Но если ты пока не справишься с такой задачей - при следующей проверке я дам более точную подсказку.



<div class="alert alert-block alert-danger">
<h2><s> Комментарий ревьюера <a class="tocSkip"></h2>
  
<b>На доработку🤔:</b>
    
Предлагаю начать проект с заголовка.

(в названии проекта лучше отразить его суть. Например, здесь можем написать что-то типа «Анализ игровых платформ».)

В самом начале важно базово проговорить контекст проекта и его цель. Это введет потенциального читателя работы в курс дела. Представь, что ты открываешь документ, а он сразу начинается с кода - это сложно воспринимать. Работу будем показывать, например, в портфолио будущему работодателю. Ну и это просто правило хорошего тона − объяснять потенциальному читателю, что здесь происходит.

 


Анализ игровых платформ.
Проект посвящен анализу данных о продажах компьютерных игр для выявления закономерностей,
определяющих успешность игры.

<div class="alert alert-warning">
<h2> Комментарий ревьюера  #2 <a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> 

Заголовок можем оформить крупным шрифтом для заголовка. Используй теги `<h>`
       
</div>



In [1]:
import pandas as pd
import numpy as np

try:
    df = pd.read_csv(r'C:\Users\glebi\Downloads\new_games.csv')
except:
    df = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')
print("Первые 5 строк данных:")
print(df.head())

print("\nИнформация о данных:")
print(df.info())

Первые 5 строк данных:
                       Name Platform  Year of Release         Genre  NA sales  \
0                Wii Sports      Wii           2006.0        Sports     41.36   
1         Super Mario Bros.      NES           1985.0      Platform     29.08   
2            Mario Kart Wii      Wii           2008.0        Racing     15.68   
3         Wii Sports Resort      Wii           2009.0        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing     11.27   

  EU sales JP sales  Other sales  Critic Score User Score Rating  
0    28.96     3.77         8.45          76.0          8      E  
1     3.58     6.81         0.77           NaN        NaN    NaN  
2    12.76     3.79         3.29          82.0        8.3      E  
3    10.93     3.28         2.95          80.0          8      E  
4     8.89    10.22         1.00           NaN        NaN    NaN  

Информация о данных:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0

<div class="alert alert-block alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<h2><s> Комментарий ревьюера <a class="tocSkip"></h2>
  
<b>На доработку🤔:</b>
 
Важно предусмотреть возможность загрузки датасета в среде Практикума. Это наша основная среда, где мы загружаем и проверяем проект. Проект должен отрабатывать в этой среде без падений в коде.
    
Такая загрузка для нас здесь будет правильной:
    
`df = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')`
    
Ссылки на датасеты всегда можно найти в описании задания проекта.


Есть удобный способ загрузки датасетов, чтобы они у нас отрабатывали и локально, и в Практикуме. Нужно использовать конструкцию try-except
      
   
```
try:
    df = pd.read_csv('new_games.csv')
except:
    df = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')
```

В таком виде мы сначала пробуем загрузить датасет с домашнего компьютера (ну или как-то иначе локально), а если не выходит, интерпретатор python перейдет к запасному варианту загрузки из хранилища. 


</div> 
 
       

После получения данных,было сделано несколько выводов: 
1)Мы получили данные за большой период времени(как минимум с 1985 года),всего 16956 игр на различных платформах.
2)Данные соотвествуют описанию,однако в столбцах встречаются ошибки в виде нечисловых значений в столбцах,у которых должен быть числовый тип данных(EU,JP sales представлены в виде типа данных object из-за встречающихся неправильных значений по типу 'tbh',которые следует заменить на NaN и сооотвественно перевести столбцы в нужный тип данных).
3)Пропуски в большом количестве в столбцах с оценкой(теоретически может быть обьяснено тем,что на старые игры может и не быть оценки.вероятнее всего эти данные отпадут после фильтрации и не будут нужны)

<div class="alert alert-warning">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Некоторые замечания и рекомендации💡:</b> 


Чтобы пункты в описании не слипались в одну строку, можно делать пробелы между строками в редакторе (при написании) или оформить пунктами:
    
* раз
* два
* три


<div class="alert alert-success">
<h2> Комментарий ревьюера  <a class="tocSkip"> </h2>

👍 Здесь все верно по выявленным проблемам в данных. Можно также обратить внимание на несоответствия стилю названий колонок.
 
   


In [2]:
print("Текущие названия столбцов:")
print(df.columns.tolist())

df.columns = df.columns.str.lower().str.replace(' ', '_')

print("\nНовые названия столбцов:")
print(df.columns.tolist())

Текущие названия столбцов:
['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales', 'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating']

Новые названия столбцов:
['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales', 'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating']


<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

👍 Отлично по переименованию.  Замечательно, что делаешь перепроверку результатов.
    
Для приведения к змеиному стилю, чтобы не делать вручную, можно также использовать такой вариант:


`df.columns = df.columns.str.lower().str.replace(' ','_')`
   

</div>

Названия столбцов были приведены с более правильному и удобному виду.

<div class="alert alert-block alert-danger" style="border-radius: 15px; box-shadow: 4px 4px 4px; border: 1px solid ">
<h2> <s>Комментарий ревьюера <a class="tocSkip"></h2>
  
<b>На доработку🤔:</b>

По заданию проекта нам важно работать с изначальным датасетом вплоть до окончания этапа предобработки. Фильтрацию сделаем после шага предобработки данных.
    
Будем считать, что предобработку мы делаем не только для себя (но и для коллег), и не только под одну аналитическую задачу.



 <div class="alert alert-block alert-danger">
<h2><s> Комментарий ревьюера <a class="tocSkip"></h2>
  
<b>На доработку🤔:</b>
 
Сейчас у нас нарушена последовательность действий. Помним о принципе GIGO - сначала данные предобрабатываем, убираем из них весь мусор, исправляем ошибки, затем данные модифицируем (добавляем необходимое для анализа, например, категоризацию), затем анализируем. Делать это в обратном порядке не стоит.
    
***
    
Более подробно, порядок действий у нас должен быть таким:
    
* описание проекта
* загрузка импортов и данных
* анализ типов данных. Здесь нужно будет указать, какие типы данных не подходят под содержание столбцов (object, хотя в столбце числа, например).
* проверка стиля столбцов (нужен змеиный стиль)
* меняем тип данных на численный там, где необходимо (в трех колонках)
* считаем количество и долю пропусков в данных.
* комментируем, где в данных пропуски, аргументируем каждый случай заполнения пропусков
* заполняем пропуски (для региональных продаж нас просят заполнить пропуски средним с группировкой по году и платформе)
* проверяем, что в данных не осталось пропусков.
* финально можно привести некоторые колонки к целочисленному типу (там где у нас float, хотя по сути может быть целое число. Например, год выпуска игры).
* изучаем неявные дубликаты в данных. Обрати внимание на категории возрастного рейтинга. Некоторые категории могут дублировать друг друга по смыслу. Также обрати внимание на названия игр. Некоторые могут дублироваться, но быть по-разному написанными (с маленькой и большой буквы).
* исправляем неявные дубликаты
* изучаем (выводим количество дубликатов) и исправляем явные дубликаты. 
* замеряем, какую долю данных мы удалили в результате всей предобработки.
* пишем промежуточный вывод по предобработке.
* фильтруем данные по необходимому периоду.
* делаем категоризацию рейтингов
* выводим топ-7 платформ 
* пишем финальный вывод.
    
***
    
Полностью проверить проект смогу после восстановления предложенной в задании схемы и последоательности действий.

In [3]:
for col in ['critic_score', 'user_score', 'eu_sales', 'jp_sales']:
    df.loc[:, col] = pd.to_numeric(df[col], errors='coerce')

print(df[['critic_score', 'user_score', 'eu_sales', 'jp_sales']].dtypes)
print("\nПервые несколько строк после преобразования:")
print(df[['critic_score', 'user_score', 'eu_sales', 'jp_sales']].head(10))

critic_score    float64
user_score      float64
eu_sales        float64
jp_sales        float64
dtype: object

Первые несколько строк после преобразования:
   critic_score  user_score  eu_sales  jp_sales
0          76.0         8.0     28.96      3.77
1           NaN         NaN      3.58      6.81
2          82.0         8.3     12.76      3.79
3          80.0         8.0     10.93      3.28
4           NaN         NaN      8.89     10.22
5           NaN         NaN      2.26      4.22
6          89.0         8.5      9.14      6.50
7          58.0         6.6      9.18      2.93
8          87.0         8.4      6.94      4.70
9           NaN         NaN      0.63      0.28


<div class="alert alert-success">
<h2> Комментарий ревьюера #2 <a class="tocSkip"> </h2>

👍 Верно

Все нечисловые значение заменили на NaN

In [4]:
missing_abs = df.isna().sum()
print("Количество пропусков в каждом столбце:")
print(missing_abs)

missing_percent = (df.isna().sum() / len(df)) * 100
print("\nПропуски в %:")
print(missing_percent.round(2))

print("\nСтолбцы с пропусками (отсортировано):")
print(missing_percent[missing_percent > 0].sort_values(ascending=False).round(2))

Количество пропусков в каждом столбце:
name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

Пропуски в %:
name                0.01
platform            0.00
year_of_release     1.62
genre               0.01
na_sales            0.00
eu_sales            0.04
jp_sales            0.02
other_sales         0.00
critic_score       51.39
user_score         54.66
rating             40.52
dtype: float64

Столбцы с пропусками (отсортировано):
user_score         54.66
critic_score       51.39
rating             40.52
year_of_release     1.62
eu_sales            0.04
jp_sales            0.02
name                0.01
genre               0.01
dtype: float64


<div class="alert alert-success">
<h2> Комментарий ревьюера 2<a class="tocSkip"> </h2>
 
👍  Можно вывести всю таблицу разом



```
missing_table = pd.DataFrame({
    'Количество пропусков': df.isnull().sum(),
    'Доля пропусков (%)': (df.isnull().mean() * 100).round(2)
})

missing_table = missing_table.sort_values('Количество пропусков', ascending=False)
missing_table
```



 Аргументация:
    Небольшое количество пропусков в eu_sales (6) и jp_sales (4) скорее всего связано с неполнотой данных,а не с отсутствием продаж. Заполнение нулями сильно исказило бы реальную картину, создав ложное
впечатление, что эти игры совсем не продавались в регионе. Более обоснованно предположить,что продажи этих игр были близки к средним показателям для аналогичных игр того же года и платформы.
Поэтому заполняем пропуски средними значениями по группе (год + платформа), а если группа не найдена - общим средним по региону.
    Для рейтингов ситуация иная: огромное количество пропусков (6871) связано с тем, что многие старые
или малоизвестные игры действительно не получали оценок от критиков. Такие данные лучше оставить как есть
или явно обозначить как "Unknown" для сохранения возможности анализа зависимости продаж от наличия оценок.




<div class="alert alert-block alert-danger">
<h2> <s>Комментарий ревьюера <a class="tocSkip"></h2>
  
<b>На доработку🤔:</b>

По коду ниже верно, а по аргументации выше - нет. Важно, чтобы код и комментарии сочетались друг с другом.

Просто заполнить региональные продажи нулем будет плохим решением, так слишком сильно исказим данные. Отсутствие информации не говорит об отсутствии продаж. Примерно как отсутствие ценника на товаре в магазине не говорит, что товар бесплатный.


В задании нам предлагается заполнить здесь пропуски средним, но с группировкой по году и платформе. 


Ок, кажется, что все-таки ситуация, что на месте пропусков у нас нулевые продажи, маловероятна. Более вероятна ситуация, что наши данные неполные. Тогда стоит посмотреть, у нас пропуски в региональных продажах идут в годах, когда в основном продавались консоли, а игры к ним были приложениями и стоили примерно одинаково. Игр было мало и многие, кто покупал консоли, покупали большинство игр. Продажи игр внутри одного года и консили были примерно одинаковыми. А вот продажи в среднем по региону без учета платформы и года могли отличаться на порядки.



In [5]:
print("Пропусков ДО заполнения:")
print(df[['eu_sales', 'jp_sales', 'rating']].isna().sum())
print()

# 1. Заполнение пропусков в продажах средним по группе
for col in ['eu_sales', 'jp_sales']:
    # Для каждого уникального сочетания года и платформы
    for (year, platform), group in df.groupby(['year_of_release', 'platform']):
        # Считаем среднее по этой группе
        group_mean = group[col].mean()
        if pd.notna(group_mean):
            # Заполняем пропуски в этой группе
            mask = (df['year_of_release'] == year) & (df['platform'] == platform) & (df[col].isna())
            df.loc[mask, col] = group_mean
    
    # Оставшиеся пропуски заполняем общим средним
    df[col] = df[col].fillna(df[col].mean())

# 2. Заполнение рейтинга
df['rating'] = df['rating'].fillna('Unknown')

print("Пропусков ПОСЛЕ заполнения:")
print(df[['eu_sales', 'jp_sales', 'rating']].isna().sum())

Пропусков ДО заполнения:
eu_sales       6
jp_sales       4
rating      6871
dtype: int64

Пропусков ПОСЛЕ заполнения:
eu_sales    0
jp_sales    0
rating      0
dtype: int64



<div class="alert alert-success">
<h2> Комментарий ревьюера  2<a class="tocSkip"> </h2>

👍  По действиям все верно


In [6]:
print("ПРИВЕДЕНИЕ ГОДА К ЦЕЛОЧИСЛЕННОМУ ТИПУ")

print(f"Тип данных year_of_release ДО: {df['year_of_release'].dtypes}")

df['year_of_release'] = df['year_of_release'].astype('Int64')

print(f"Тип данных year_of_release ПОСЛЕ: {df['year_of_release'].dtypes}")
print(f"Диапазон годов: {df['year_of_release'].min()} - {df['year_of_release'].max()}")

ПРИВЕДЕНИЕ ГОДА К ЦЕЛОЧИСЛЕННОМУ ТИПУ
Тип данных year_of_release ДО: float64
Тип данных year_of_release ПОСЛЕ: Int64
Диапазон годов: 1980 - 2016


In [7]:
print("Уникальные платформы :")
print(df['platform'].unique())

print("\nУникальные жанры:")
print(df['genre'].unique())

print("\nУникальные рейтинги:")
print(df['rating'].unique())

Уникальные платформы :
['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' 'PS4' '3DS' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'WiiU' 'GC' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']

Уникальные жанры:
['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'
 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy' nan 'MISC'
 'ROLE-PLAYING' 'RACING' 'ACTION' 'SHOOTER' 'FIGHTING' 'SPORTS' 'PLATFORM'
 'ADVENTURE' 'SIMULATION' 'PUZZLE' 'STRATEGY']

Уникальные рейтинги:
['E' 'Unknown' 'M' 'T' 'E10+' 'K-A' 'AO' 'EC' 'RP']


В жанрах встречаются неявные дубликаты,жанры одинаковые,но регистры разные

In [8]:
# Приводим к нижнему регистру
df['genre'] = df['genre'].str.lower()

print("\nУникальные жанры ПОСЛЕ:")
print(df['genre'].unique())
print(f"Количество уникальных жанров ПОСЛЕ: {df['genre'].nunique()}")


Уникальные жанры ПОСЛЕ:
['sports' 'platform' 'racing' 'role-playing' 'puzzle' 'misc' 'shooter'
 'simulation' 'action' 'fighting' 'adventure' 'strategy' nan]
Количество уникальных жанров ПОСЛЕ: 12


In [9]:
# 2. Названия игр - приводим к нижнему регистру
print("2. НАЗВАНИЯ ИГР:")
print(f"Уникальные названия ДО: {df['name'].nunique()}")
df['name'] = df['name'].str.lower()
print(f"Уникальные названия ПОСЛЕ: {df['name'].nunique()}")
print()

# 3. Возрастные рейтинги - проверяем (обычно дубликатов нет, но оставляем как есть)
print("3. ВОЗРАСТНЫЕ РЕЙТИНГИ:")
print(f"Уникальные рейтинги: {df['rating'].unique()}")
print(f"Количество: {df['rating'].nunique()}")

2. НАЗВАНИЯ ИГР:
Уникальные названия ДО: 11559
Уникальные названия ПОСЛЕ: 11559

3. ВОЗРАСТНЫЕ РЕЙТИНГИ:
Уникальные рейтинги: ['E' 'Unknown' 'M' 'T' 'E10+' 'K-A' 'AO' 'EC' 'RP']
Количество: 9


Все жанры,игры и возрастной рейтинг привели
привели к единому виду для избежания дубликатов

In [10]:
duplicates = df.duplicated().sum()
print(f"Количество явных дубликатов: {duplicates}")

if duplicates > 0:
    df = df.drop_duplicates()
    print(f"После удаления осталось {len(df)} строк")
else:
    print("Дубликатов нет, можно продолжать")

Количество явных дубликатов: 241
После удаления осталось 16715 строк



<div class="alert alert-success">
<h2> Комментарий ревьюера 2 <a class="tocSkip"> </h2>

👍  

Избавились от явных дубликатов в датафрейме

In [11]:
print("ДОЛЯ УДАЛЕННЫХ ДАННЫХ")

initial_rows = 16956  
final_rows = len(df)
deleted_rows = initial_rows - final_rows
deleted_percent = (deleted_rows / initial_rows) * 100

print(f"Исходное количество строк: {initial_rows}")
print(f"Количество строк после предобработки: {final_rows}")
print(f"Удалено строк: {deleted_rows} ({deleted_percent:.1f}%)")

ДОЛЯ УДАЛЕННЫХ ДАННЫХ
Исходное количество строк: 16956
Количество строк после предобработки: 16715
Удалено строк: 241 (1.4%)



<div class="alert alert-success">
<h2> Комментарий ревьюера 2 <a class="tocSkip"> </h2>

👍  

В ходе предобработки выполнено:

1. Приведены названия столбцов к змеиному стилю
2. Преобразованы типы данных в столбцах с оценками и продажами
3. Обработаны пропуски:
   - Продажи: заполнены средними по группе (год + платформа)
   - Рейтинг: заполнен 'Unknown'
   - Оценки: оставлены как NaN
4. Устранены неявные дубликаты в жанрах (приведены к нижнему регистру)
5. Удалено явных дубликатов: 182
6. Общая потеря данных: 1.1%

Данные готовы к фильтрации и анализу.


<div class="alert alert-success">
<h2> Комментарий ревьюера 2 <a class="tocSkip"> </h2>

👍  Здесь хорошо по оформлению промежуточного вывода.

Отфильтруемм данные по годам,нас интересует периол с 2003 по 2012

In [12]:
df = df[(df['year_of_release'] >= 2000) & (df['year_of_release'] <= 2013)].copy()

print(f"Стало строк после фильтрации: {len(df)}")
print(f"Диапазон годов: {df['year_of_release'].min()} - {df['year_of_release'].max()}")


Стало строк после фильтрации: 12781
Диапазон годов: 2000 - 2013



<div class="alert alert-success">
<h2> Комментарий ревьюера 2 <a class="tocSkip"> </h2>

👍  Замечательно по перепроверке.

In [13]:
#user_score
def categorize_user_score(score):
    if pd.isna(score):
        return 'нет оценки'
    elif score >= 8 and score <= 10:
        return 'высокая'
    elif score >= 3 and score < 8:
        return 'средняя'
    elif score >= 0 and score < 3:
        return 'низкая'
    else:
        return 'некорректная'

df['user_score_category'] = df['user_score'].apply(categorize_user_score)

#critic_score
def categorize_critic_score(score):
    if pd.isna(score):
        return 'нет оценки'
    elif score >= 80 and score <= 100:
        return 'высокая'
    elif score >= 30 and score < 80:
        return 'средняя'
    elif score >= 0 and score < 30:
        return 'низкая'
    else:
        return 'некорректная'

df['critic_score_category'] = df['critic_score'].apply(categorize_critic_score)

print("Распределение по категориям пользовательских оценок:")
print(df['user_score_category'].value_counts())
print("\nРаспределение по категориям оценок критиков:")
print(df['critic_score_category'].value_counts())

Распределение по категориям пользовательских оценок:
нет оценки    6298
средняя       4081
высокая       2286
низкая         116
Name: user_score_category, dtype: int64

Распределение по категориям оценок критиков:
нет оценки    5612
средняя       5422
высокая       1692
низкая          55
Name: critic_score_category, dtype: int64



<div class="alert alert-success"> 
<h2> Комментарий ревьюера 2<a class="tocSkip"> </h2>

👍 Ок. Альтернативно мы могли бы использовать pd.cut(), но брать в категории значения по нижней границе (для этого нужно использовать аргумент `right=False`)
    
`pd.cut(df_actual['user_score'], bins=[0, 3, 8, 11], labels=["низкая оценка", "средняя оценка", "высокая оценка"]), right=False)`





Анализируя результаты распределения оценок можно сделать несколько выводов:
1) Большинство игр не имееют какой либо оценки
2) Пользователи более эмоциональны и предвзяты,они реже ставят средние оценки,и чаще высокие и низкие,по сравнению с критиками

In [14]:
top_platforms = df['platform'].value_counts().head(7)
print("Топ-7 платформ по количеству игр (2000-2013):")
print(top_platforms)

Топ-7 платформ по количеству игр (2000-2013):
PS2     2127
DS      2120
Wii     1275
PSP     1180
X360    1121
PS3     1087
GBA      811
Name: platform, dtype: int64



<div class="alert alert-success"> 
<h2> Комментарий ревьюера 2<a class="tocSkip"> </h2>

👍 Здесь все корректно.
</div>  
	  




В период с 2000 по 2013 год лидером по количеству тайтлов является PS2, сразу за ней ,с небольшим отставанием в 7 проектов идет Nintendo DS.
Все остальные платформы отстают гораздо сильнее. Из топ-7 только 6 платформ перевалили за 1000 игр.В топ-7 доминируют Nintendo и PS,а Xbox представлен лишь одной платформой - X360

В ходе проекта был проведен анализ данных о продажах компьютерных игр. Исходный датафрейм содержал 16956 строк. После предобработки данных (преобразование типов, заполнение пропусков, приведение жанров и названий к нижнему регистру, удаление дубликатов) осталось 16774 строк, потеря данных составила 1,1%.
Для анализа был выбран актуальный период с 2000 по 2013 год, в который попало 8820гр.
Топ-7 платформ по количеству выпущенных игр:
- PS2 — 2131 игра
- DS — 2126 игр
- Wii — 1279 игр
- PSP — 1184 игры
- X360 — 1126 игр
- PS3 — 1090 игр
- GBA — 813 игр

В период с 2000 по 2013 год лидером по количеству тайтлов является PS2, сразу за ней с небольшим отставанием идет Nintendo DS. Все остальные платформы отстают гораздо сильнее. Из топ-7 только 6 платформ перевалили за 1000 игр. В топ-7 доминируют Nintendo и Sony, а Xbox представлен лишь одной платформой — X360.
Анализ распределения оценок показал:
- Большинство игр не имеют какой-либо оценки (как пользовательской, так и от критиков)
- Пользователи более эмоциональны и предвзяты — они реже ставят средние оценки и чаще высокие или низкие по сравнению с критиками

Данные успешно предобработаны и готовы к дальнейшему анализу для планирования рекламных кампаний на 2017 год.

 <div class="alert alert-block alert-danger">
<h2><s> Комментарий ревьюера 2<a class="tocSkip"></h2>
  
<b>На доработку🤔:</b>
    
Убедись, что проект отрабатывает до самого конца без ошибок в коде.
    
Текстовые комментарии и выводы лучше просто записать в текстовых ячейках, так будет меньше визуального шума. Текст в коде нам не нужен.


<div style="border:solid Chocolate 2px; padding: 40px">

**Общий вывод по проверке**

Сейчас пришел проект с нарушенной схемой действий. Прошу тебя выполнить все описанные в задании шаги в предложенном в задании порядке. Полностью проверить работу смогу после восстановления схемы проекта.
    
**Желаю удачи и жду твой проект на повторное ревью!**😉

</div>

<div style="border:solid Chocolate 2px; padding: 40px">

**Комментарий ко второй проверке**

Спасибо за качественную доработку. Практически все верно. Однако еще кое-что нужно поправить.

* Важно уточниться по аргументации заполнения пропусков в региональных продажах
* В конце проекта выводы стоит написать текстом, а не в кодовой ячейке и убедиться, что код проекта отрабатывает целиком и не падает.
    
**Желаю удачи и жду твой проект на повторное ревью!**😉
</div>



<div style="border:solid Chocolate 2px; padding: 40px">

**Комментарий к финальной проверке**

Спасибо за доработку. Критичных комментариев нет. 

Если что-то пропустил или остались вопросы, напиши в телеграм @kriill

Желаю удачи в дальнейшей учебе!

</div>